disperser/dataapi/queried_operators_handlers.go

```diff
@@ -198,7 +198,11 @@ func checkIsOnlineAndProcessOperator(operatorStatus OperatorOnlineStatus, operat
 	var isOnline bool
 	var socket string
 	if operatorStatus.IndexedOperatorInfo != nil {
-		socket = core.OperatorSocket(operatorStatus.IndexedOperatorInfo.Socket).GetV1RetrievalSocket()
+		operatorSocket, err := core.ParseOperatorSocket(operatorStatus.IndexedOperatorInfo.Socket)
+		if err != nil {
+			logger.Warn("Failed to parse operator socket", "err", err)
+		}
+		socket = operatorSocket.GetV1RetrievalSocket()
```

Comment:
If `err != nil`, how can this still get v1 retrieval socket?

Reply:
There is an existing implementation issue where the expectation is to call get sockets method to return the zero value on error conditions. This will be a longer pull refactor.

Ideally we should not rely on zero values as errors, instead anytime we get an *OperatorSocket struct and an error we need to handle error and return before calling its implemented methods. Will get this addressed in the next PR

Reply:
Here as well